starter-code/js/queue.js

```diff
@@ -0,0 +1,60 @@
+btnAdd = document.getElementById("add-queue");
+btnTake = document.getElementById("take-queue");
+
+btnAdd.onclick = function() {
+    queueInitial = new QueueDataStructure();
+    queueInitial=getQueue();
+
+    let queueFinal = queueInitial.enqueue(document.getElementById("queue-input").value);
+
+    if (typeof queueFinal === "string") {
+        alert(queueFinal)
+    }
+    else {displayQueue(queueInitial);}
+}
+
+btnTake.onclick = function() {
+    queueInitial = new QueueDataStructure();
+    queueInitial=getQueue();
+
+    let queueFinal = queueInitial.dequeue(document.getElementById("queue-input").value);
+
+    if (queueFinal === 'Queue Underflow') {
+        alert(queueFinal)
+    }
+    else {
+        queueInitial.queueControl.pop();
+        displayQueue(queueInitial);
```

Comment:
Here you are using the pop method on the queueControl property, you have created methods to interact with your queue